Release/include/pplx/threadpool.h

```diff
@@ -70,6 +72,10 @@ class threadpool
     threadpool(size_t num_threads) : m_service(static_cast<int>(num_threads)) {}
 
     boost::asio::io_service m_service;
+
+private:
+    static bool m_threads_started;
```

Comment:
Do we really need a separate bool for this? IMHO we could reuse `m_num_threads`, i.e. initialize it to 0 and set it to the default value when the thread pool is actually created.

Reply:
@vadz easy enough to do any way. I think at the time (I originally did this a year ago and have it working in my code base for awhile) I just wanted more separation.

To do as you propose, we still need another const to hold the default value. To me this is slightly more clear. But more than happy to change this if people feel otherwise.

Reply:
@vadz hmm, maybe my approach is wrong, but it ends up making the code look clunkier. 

This means that every time I instantiate the static s_shared, I have to do this:

  ```
    if (!threadpool::m_num_threads) {
        threadpool::m_num_threads = default_num_threads;
    }
    static threadpool_impl s_shared(threadpool::m_num_threads);
```

I can't change the constructor because it needs to call the `crossplat::threadpool(n)` in the initialization list.

This is replicated 3 times for the 3 occurrence and hence (to me) seems clunkier than

`
    static threadpool_impl s_shared(threadpool::m_num_threads);
    threadpool::m_threads_started = true;
`

Thoughts?

Reply:
Another option would be this.

`
    static threadpool_impl s_shared(threadpool::m_num_threads ? threadpool::m_num_threads : default_num_threads);
`

where 

`
static const size_t default_num_threads = 40;
`

I myself tend not to use this syntax, but would seem to work well here. Although it would still need to be replicated for each shared_instance version.

Reply:
Sorry, I'm not sure what the problem is exactly but, just in case it can be useful, the way I'd do it would be like this:

1. Remove `threadpool_impl` ctor argument, if only to avoid repeating it thrice as in the current code.
2. Add `threadpool::get_num_threads()` that would set `m_num_threads` to the default value if it's 0 and return it.
3. Use `threadpool::get_num_threads()` to `crossplat::threadpool` ctor in `threadpool_impl` ctor initializer list.

Reply:
The reason I asked was to try and limit the impact of the code. Since I cannot claim to be overly familiar with all the coding conventions, I thought it best to ask versus add stuff that doesn't stay aligned with the coding conventions.

BTW, I did as you recommended however there is one sticking point, at least for me. While I can certainly pass ```get_num_threads()``` to ```threadpool``` ctor, the num of threads is also used int the body of the construct. Hence There is code like this:

```
    {
        for (size_t i = 0; i < threadpool::get_num_threads(); i++)
                   add_thread();
    }   
```

Not the end of the world, but I guess I'm not fond of the dual calls on ```get_num_threads```. 

Anyways, I'll commit the code shortly and you can comment further. Happy to change it in anyway. 

Usually I error on the side of caution until I'm way more familiar with the code set ... so please bear with the questions.